master/buildbot/changes/bitbucket.py

```diff
@@ -100,75 +100,71 @@ def _getChanges(self):
-                if not current or current[0:12] != revision[0:12]:
-                    # parse pull request api page (required for the filter)
-                    response = yield self._getPage(str(pr['links']['self']['href']).replace('!api', 'api'))
-                    page = yield readBody(response)
-                    pr_json = json.loads(page, encoding=self.encoding)
-
-                    # filter pull requests by user function
-                    if not self.pullrequest_filter(pr_json):
-                        log.msg('pull request does not match filter')
-                        continue
-
-                    # access additional information
-                    author = pr['author']['display_name']
-                    prlink = pr['links']['html']['href']
-                    # Get time updated time. Note that the timezone offset is
-                    # ignored.
-                    if self.useTimestamps:
-                        updated = datetime.strptime(
-                            pr['updated_on'].split('.')[0],
-                            '%Y-%m-%dT%H:%M:%S')
-                    else:
-                        updated = epoch2datetime(reactor.seconds())
-                    title = pr['title']
-                    # parse commit api page
-                    response = yield self._getPage(str(pr['source']['commit']['links']['self']['href']).replace('!api', 'api'))
-                    page = yield readBody(response)
-                    commit_json = json.loads(page, encoding=self.encoding)
-                    # use the full-length hash from now on
-                    revision = commit_json['hash']
-                    revlink = commit_json['links']['html']['href']
-                    # parse repo api page
-                    response = yield self._getPage(str(pr['source']['repository']['links']['self']['href']).replace('!api', 'api'))
-                    page = yield readBody(response)
-                    repo_json = json.loads(page, encoding=self.encoding)
-                    repo = repo_json['links']['html']['href']
-
-                    # update database
-                    yield self._setCurrentRev(nr, revision)
-                    # emit the change
-                    yield self.master.data.updates.addChange(
-                        author=ascii2unicode(author),
-                        revision=ascii2unicode(revision),
-                        revlink=ascii2unicode(revlink),
-                        comments=u'pull-request #%d: %s\n%s' % (
-                            nr, title, prlink),
-                        when_timestamp=datetime2epoch(updated),
-                        branch=ascii2unicode(branch),
-                        category=self.category,
-                        project=self.project,
-                        repository=ascii2unicode(repo),
-                        src=u'bitbucket',
-                    )
+        if 'values' in result:
```

Comment:
If not 'values' in result:
return
And you avoid an extra indentation, or result.get('values', [])